src/session_impl.cpp

```diff
@@ -594,16 +594,23 @@ namespace aux {
 	void session_impl::load_state(bdecode_node const* e
 		, std::uint32_t const flags)
 	{
+		settings_pack sp;
+		load_state_impl(*e, flags, sp, true);
+	}
+
+	void session_impl::load_state_impl(bdecode_node const& e
```

Comment:
I feel like perhaps it would be cleaner if there was a free function that could load state from a file into a ``session_params`` object, and make sure all state can fit in there.